File: projects/rpn2tex/migrations_archive/go-feature-by-feature-1/token.go
package rpn2tex

import "fmt"

// TokenType represents the type of a lexical token.
type TokenType int

const (
	TokenNumber TokenType = iota
	TokenPlus
	TokenMinus
	TokenTimes
	TokenDivide
	TokenEOF
)

// Token represents a lexical token with position information.
type Token struct {
	Type   TokenType
	Value  string
	Line   int // 1-based
	Column int // 1-based
}

// String returns a string representation of the token for debugging.
func (t Token) String() string {
	return fmt.Sprintf("Token(%s, %q, %d, %d)", t.Type, t.Value, t.Line, t.Column)
}

// String returns a string representation of the token type.
func (tt TokenType) String() string {
	switch tt {
	case TokenNumber:
		return "NUMBER"
	case TokenPlus:
		return "PLUS"
	case TokenMinus:
		return "MINUS"
	case TokenTimes:
		return "TIMES"
	case TokenDivide:
		return "DIVIDE"
	case TokenEOF:
		return "EOF"
	default:
		return "UNKNOWN"
	}
}

File: projects/rpn2tex/migrations/java-feature-by-feature-2/src/main/java/com/rpn2tex/Token.java
package com.rpn2tex;

/**
 * Immutable token produced by the lexer.
 *
 * @param type   The type of this token
 * @param value  The string representation of the token
 * @param line   1-based line number in source
 * @param column 1-based column number in source
 */
public record Token(TokenType type, String value, int line, int column) {
}
